### PR TITLE
Add Remote Bazel docs

### PR DESCRIPTION
**Pull request by maggie-lou**
**Related issues**: N/A

### CODE REVIEW

File: docs/remote-bazel-introduction.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+CI product, BuildBuddy Workflows. However, Remote Bazel is a much more flexible
+entrypoint to our CI platform.
+
+Workflows must be configured with a config YAML that is checked in to GitHub.
+Despite the performance benefits of running on Workflows, many companies have
+legacy CI workflows that would be challenging and error-prone
+to migrate to a new CI platform.
+
+Remote Bazel, on the other hand, is intended as a drop-in solution that should be
+easy to integrate into pre-existing CI pipelines. It can
```
Comment (tylerwilliams) [inline docs/remote-bazel-introduction.md:45]: Remote Bazel is a drop-in solution that can be more easily integrated into pre-existing CI pipelines.

File: docs/remote-bazel-introduction.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
+---
+id: remote-bazel-introduction
+title: Introduction to Remote Bazel
+sidebar_label: Remote bazel introduction
+---
+
+Remote Bazel is an easily configurable way to run commands on a remote runner. You
+can think of it as dynamically spinning up a VM to execute a single command.
```
Comment (tylerwilliams) [inline docs/remote-bazel-introduction.md:8]: is it only single? (don't we have steps in the api now?)

**Review by tylerwilliams — APPROVED** [review]

File: docs/remote-bazel-introduction.md (hunk 1 above)
Comment (maggie-lou) [reply docs/remote-bazel-introduction.md:45]: Much better. Thanks!

File: docs/remote-bazel-introduction.md (hunk 2 above)
Comment (maggie-lou) [reply docs/remote-bazel-introduction.md:8]: Changed the wording a bit

**Pull request merged**